src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,191 @@
+        Node<T> node = head;
+        for (int i = 0; i < index; i++) {
+            node = node.next;
+        }
+        return node.item;
     }
 
     @Override
     public T set(T value, int index) {
-        return null;
+        validIndexCheck(index);
+        T oldValue;
+        if (index == 0) {
+            oldValue = head.item;
+            head.item = value;
+            return oldValue;
+        }
+        if (index == size - 1) {
+            oldValue = tail.item;
+            tail.item = value;
+        } else {
+            Node node = new Node(head, value, head.next);
+            for (int i = 1; i < index; i++) {
+                node.prev = node.next;
+                node.next = node.next.next;
+            }
+            oldValue = (T) node.next.item;
+            node.next.item = value;
+        }
+        return oldValue;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        validIndexCheck(index);
+        Node<T> node = new Node<>(null,head.item, head.next);
+        T removedValue;
+        if (size == 1) {
+            removedValue = head.item;
+            head.item = null;
+            tail.item = null;
+        } else {
+            if (index == 0) {
+                removedValue = head.item;
+                head = head.next;
+                head.prev = null;
+                size--;
+                return removedValue;
+            }
+            if (index == size - 1) {
+                removedValue = tail.item;
+                tail = tail.prev;
+                tail.next = null;
+            } else {
+                for (int i = 0; i < index; i++) {
+                    node.prev = node.next;
+                    node.next = node.next.next;
+
+                }
+                node.item = node.prev.item;
+                removedValue = node.item;
+                node.item = null;
+                node.next.prev = node.prev.prev;
+                node.prev.prev.next = node.next;
+                node.prev = null;
+                node.next = null;
+            }
+        }
+        size--;
+        return removedValue;
     }
 
     @Override
     public boolean remove(T object) {
-        return false;
+        Node<T> node = new Node<>(null,head.item, head.next);
+        int index = 0;
+        found: {
+                for (; index < size; index++) {
+                    if (node.item == null) {
+                        if (object == null) {
+                            break found;
+                        }
+                    } else {
+                        if (node.item.equals(object)) {
+                            break found;
+                        }
+                    }
+                    if (node.next != null) {
+                        node.item = node.next.item;
+                        node.next = node.next.next;
+                    }
+                }
+            return false;
+        }
+        remove(index);
+        return true;
```

Comment:
Don't forget that you can have `null` as a method parameter in the `remove(T object)` method. Make sure your logic correctly handles the case when `object` is `null`.